verification/cbmc/proofs/aws_cryptosdk_private_derive_key/aws_cryptosdk_private_derive_key_harness.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/cipher.h>
+#include <aws/cryptosdk/private/cipher.h>
+#include <make_common_data_structures.h>
+
+void aws_cryptosdk_private_derive_key_harness() {
+    struct aws_cryptosdk_alg_properties *props = malloc(sizeof(*props));
+    struct content_key *content_key            = malloc(sizeof(*content_key));
+    struct data_key *data_key                  = malloc(sizeof(*data_key));
+    struct aws_byte_buf *message_id            = malloc(sizeof(*message_id));
+    struct aws_byte_buf *commitment            = malloc(sizeof(*commitment));
+
+    /* Assumptions */
+    ensure_alg_properties_attempt_allocation(props);
+    __CPROVER_assume(aws_cryptosdk_alg_properties_is_valid(props));
+
+    __CPROVER_assume(content_key != NULL);
+
+    __CPROVER_assume(data_key != NULL);
+
+    __CPROVER_assume(commitment != NULL);
+    __CPROVER_assume(aws_byte_buf_is_bounded(commitment, MAX_BUFFER_SIZE));
+    ensure_byte_buf_has_allocated_buffer_member(commitment);
+    __CPROVER_assume(aws_byte_buf_is_valid(commitment));
+
+    __CPROVER_assume(message_id != NULL);
+    __CPROVER_assume(aws_byte_buf_is_bounded(message_id, MAX_BUFFER_SIZE));
```

Comment:
You can replace this with 
```c
__CPROVER_assume(message_id != NULL, aws_byte_buf_is_bounded(message_id, MAX_BUFFER_SIZE));
```

Reply:
This is detected as an error by CBMC (too many arguments on `__CPROVER_assume`). Restoring it to the previous version. Did you mean something else?

Reply:
@adpaco-aws yes, I mean to use implies.
```c
__CPROVER_assume(IMPLIES(message_id != NULL, aws_byte_buf_is_bounded(message_id, MAX_BUFFER_SIZE)));
```